vignettes/BiobaseDevelopment.Rmd

```diff
@@ -0,0 +1,510 @@
+
+### Additional `eSet` methods
+
+-   `show`.
+
+-   `storageMode`: influence how `assayData` is stored.
+
+-   `updateObject`: update `eSet` objects to their current version.
+
+-   `validObject`: ensure that `eSet` is valid.
+
+The `validObject` method is particularly important to `eSet`, ensuring that `eSet` contains consistent structure to data.
+
+```{r}
+getValidity(getClass("eSet"))
+```
+
+The validity methods for `eSet` reflect our design goals. All `assayData` members must have identical row and column dimensions and `featureNames`. The names and numbers of samples must be the same in `assayData` and `phenoData` slots. Validity methods are defined for the classes underlying each slot as well. For instance, the validity methods for `AnnotatedDataFrame` check that variables used in `pData` are at least minimally described in `varMetadata`.
+
+## Subclasses of `eSet`
+
+*Biobase* defines three classes that extend `eSet`. `ExpressionSet` (discussed further below) is meant to contain microarray gene expression data. *SnpSet* is a preliminary class to contain SNP data; other classes in development (e.g., in *oligo*) may provide alternative implementations for SNP data. *MultiSet* is an `ExpressionSet`-like class, but without restriction on the names (though not structure) of elements in the `assayData` slot.
+
+### `ExpressionSet`
+
+Purpose:
+
+-   Contain gene expression data.
+
+Required `assayData` members.
+
+-   `exprs`, a matrix of expression values.
+
+Important methods.
+
+-   Initialization (additional details below):
+
+    ```{r eval=FALSE}
+    obj <- new("ExpressionSet", phenoData = new("AnnotatedDataFrame"), experimentData = new("MIAME"), annotation = character(),        exprs = new("matrix"))
+    ```
+
+-   `exprs(obj), exprs(obj) <- value`: get or set `exprs`; methods defined for *ExpressionSet*, *AssayData*.
+
+### *MultiSet* and *SnpSet*
+
+*MultiSet*.
+
+-   Purpose: flexibly contain a collection of expression data matricies.
+
+-   Required `assayData` members: none.
+
+*SnpSet*.
+
+-   Purpose: contain genomic SNP calls.
+
+-   Required `assayData` members: `call`, `callProbability`.
+
+# Comments on `assayData`: high-throughput data storage
+
+The `assayData` slot is meant to store high-throughput data. The idea is that the slot contains identically sized matrices containing expression or other data. All matrices in the slot must have the same dimension, and are structured so that rows represent 'features' and columns represent 'samples'. Validity methods enforce that row and column names of slot elements are identical.
+
+For technical reasons, creating instance of *AssayData* is slightly different from the way this is usually done in [R]{.sans-serif}. Normally, one creates an instance of a class with an expression like `new("ExpressionSet", ...)`, with the ... representing additional arguments. *AssayData* objects are created with
+
+`> assayDataNew("environment", elt)`
```

Comment:
We should use the code block structure for this with `eval=FALSE` so that we can remove the `>`.